added QualifyingReason enum to CombinedIdentityAndEligibilityResponse

AmarkumarJagtap opened the pull request.

AmarkumarJagtap committed:
added QualifyingReason enum to CombinedIdentityAndEligibilityResponse

YetAnotherMatt commented on common_dwp_api/src/main/java/uk/gov/dhsc/htbhf/dwp/model/QualifyingReason.java:
```diff
@@ -0,0 +1,15 @@
+
+public enum QualifyingReason {
+    UNDER_18,
+    NOT_SET;
```
As discussed with @mikeSmith08 @AmarkumarJagtap and Karen, QualifyingReason should have _all_ the values form QualifyingBenefits, plus under_18. It will be the job of the eligibility service to translate from the QualifyingBenefit that is returned BY DWP/HMRC into a QualifyingReason sent back to the claimant service. 
The EligibilityOverride applied by the claimant servcie should use QualifyingReason, not QualifyingBenefits.

AmarkumarJagtap committed:
replaced QualifyingBenefits to QualifyingReason in CombinedIdentityAndEligibilityResponse

AmarkumarJagtap committed:
added QualifyingBenefits to QualifyingReason in CombinedIdentityAndEligibilityResponse.
QualifyingBenefits will be removed from CombinedIdentityAndEligibilityResponse after implementing in QualifyingReason in eligibility and  claimant service

dwybourn commented on common_dwp_api/src/test/java/uk/gov/dhsc/htbhf/dwp/model/QualifyingReasonTest.java:
```diff
@@ -0,0 +1,23 @@
+class QualifyingReasonTest {
+
+    @ParameterizedTest
+    @CsvSource({
```
Child tax credit is missing.

AmarkumarJagtap replied:
added now in both QualifyingBenefitsTest and QualifyingReasonTest

AmarkumarJagtap committed:
added child tax credit to test cases

AmarkumarJagtap committed:
added qualifying reason to test data factory
